File: v1/kotlin/bob/src/main/kotlin/Bob.kt
object Bob {
    fun hey(input: String): String =
            if (input.isNullOrBlank()) "Fine. Be that way!"
            else if (input.any { Character.isLetter(it) } && input.toUpperCase().equals(input)) "Whoa, chill out!"
            else if (input.endsWith('?')) "Sure."
            else "Whatever."
}

File: v1/csharp/bob/Bob.cs
using System.Linq;

public static class Bob
{
    public static string Hey(string message)
    {
        if (string.IsNullOrWhiteSpace(message))
        {
            return "Fine. Be that way!";
        }

        if (message.Any(char.IsLetter) && message.ToUpperInvariant() == message)
        {
            return "Whoa, chill out!";
        }

        if (message.Trim().EndsWith("?"))
        {
            return "Sure.";
        }

        return "Whatever.";
    }
}